src/Microsoft.ML.Transforms/KeyToVectorMapping.cs

```diff
@@ -466,7 +466,9 @@ public override SchemaShape GetOutputSchema(SchemaShape inputSchema)
                         metadata.Add(new SchemaShape.Column(AnnotationUtils.Kinds.SlotNames, SchemaShape.Column.VectorKind.Vector, keyMeta.ItemType, false));
                 if (col.Kind == SchemaShape.Column.VectorKind.Scalar)
                     metadata.Add(new SchemaShape.Column(AnnotationUtils.Kinds.IsNormalized, SchemaShape.Column.VectorKind.Scalar, BooleanDataViewType.Instance, false));
-                result[colInfo.outputColumnName] = new SchemaShape.Column(colInfo.outputColumnName, SchemaShape.Column.VectorKind.Vector, NumberDataViewType.Single, false, new SchemaShape(metadata));
+                result[colInfo.outputColumnName] = new SchemaShape.Column(colInfo.outputColumnName,
+                    col.Kind == SchemaShape.Column.VectorKind.VariableVector ? SchemaShape.Column.VectorKind.VariableVector : SchemaShape.Column.VectorKind.Vector,
```

Comment:
>col [](start = 20, length = 3)

This is key-to-vector mapping. Can the input column be a vector-valued?

Reply:
You can have vector of keys, right?

---
In reply to: [263169008](https://github.com/dotnet/machinelearning/pull/2681#discussion_r263169008) [](ancestors = 263169008)

Reply:
In fact a vector of keys is one of the most important scenarios (e.g., bag of words).